packages/auth-providers/supabase/middleware/README.md

```diff
@@ -20,12 +14,24 @@ interface Props {
   meta?: TagDescriptor[]
 }
 
+type SupabaseAppMetadata = {
+  provider: string
+  providers: string[]
+  roles: string[]
+}
+
 export const registerMiddleware = () => {
   const supabaseAuthMiddleware = initSupabaseMiddleware({
     // Optional. If not set, Supabase will use its own `currentUser` function
     // instead of your app's
     getCurrentUser,
+    // Optional. If you wish to enforce RBAC, define a function to return roles.
+    // Typically, one will define roles in Supabase in the user's app_metadata.
```

Comment:
I think this might be incorrect here! getRoles needs an array of strings, but ofcourse if you setup your DB to return that it should be ok! 

Remember the default one already does this :)

Reply:
Hm interesting. I had roles set to just a string in the Supabase raw_app_metadata and roles were enforced.

I'll confirm,

Reply:
It will be, if you use the default one i.e. don't give it a "getRoles"! It checks if its a string, and will convert it to an array if required

Roles will be enforced regardless at the moment (because it doesn't use the server state, but the hasRole function).

To verify you'll need to check that roles appears as a string array in the serverContext

Reply:
Agreed. A better example is:


```ts
/**
 * Get roles from Supabase user app_metadata
 */
export const getRoles = ({
  app_metadata,
}: {
  app_metadata: SupabaseAppMetadata
}): string[] => {
  if (typeof app_metadata.roles === 'string') {
    return [app_metadata.roles]
  }

  return app_metadata.roles
}
```

And I now see how the default roles works.